.github/workflows/workflow.yml

```diff
@@ -8,105 +8,76 @@ on:
   workflow_dispatch:
 
 jobs:
-  build:
+  lint:
     runs-on: ubuntu-20.04
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+
+      - name: Install Poetry
+        run: pipx install poetry
+
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: 3.7.12 # Patch version must be specified to avoid any cache confusion, since the cache key depends on the full Python version. Any potentiel difference in patches between jobs will lead to a cache not found error.
-      - name: Cache build
-        id: restore-build
-        uses: actions/cache@v2
-        with:
-          path: ${{ env.pythonLocation }}
-          key: build-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ github.sha }} # Cache the entire build Python environment
-          restore-keys: |
-            build-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}
-            build-${{ env.pythonLocation }}-
-      - name: Build package
-        run: make build
-      - name: Cache release
-        id: restore-release
-        uses: actions/cache@v2
-        with:
-          path: dist
-          key: release-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ github.sha }}
+          cache: poetry
+          cache-dependency-path: pyproject.toml
```

Comment:
Why caching pyproject.toml ? Is it supposed to be change by CI ?
If you need to cache pyproject.toml why not caching poetry.lock ?

Reply:
I understood that by leaving `poetry.lock` out of version-control, we keep the deterministic benefits of poetry while leaving people the choice of their own versions, which would be otherwise impossible without changing the lock.

In such scenario, to use `poetry.lock` as a base for the cache key becomes impossible, or almost: we could have an extra step to create an empty lock if there is none, and cache it.

Do you know what are the common practices here?

Reply:
From https://python-poetry.org/docs/basic-usage/:

> Committing this file to VC is important because it will cause anyone who sets up the project to use the exact same versions of the dependencies that you are using.

> For libraries it is not necessary to commit the lock file.

Reply:
Well I'm not sure I understand the phrase "For libraries it is not necessary to commit the lock file." from https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control

At LexImpact we use Poetry in every Python project and we commit poetry.lock to be sure to deploy the version of the package that we have tested.

Reply:
Done.